process/block/shardblock.go

```diff
@@ -680,6 +721,12 @@ func (sp *shardProcessor) CommitBlock(
 		return err
 	}
 
+	err = sp.checkEpochCorrectnessCrossChain(chainHandler)
```

Comment:
This code should be executed now without the initial condition if header.IsStartOfEpochBlock() {  ?

Reply:
initial condition is not more needed. It is better to check every time.